theano/gof/vm.py

```diff
@@ -894,6 +894,68 @@ def make_all(self, profiler=None, input_storage=None,
             compute_map[k] = [k.owner is None]
 
         thunks = []
+
+        if self.allow_gc:
+
+            viewed_by = {}
+            for var in fgraph.variables:
+                viewed_by[var] = []
+            view_of = {}
+
+            for idx in range(len(order)):
+                node = order[idx]
+                dmap = getattr(node.op, 'destroy_map', None)
+                vmap = getattr(node.op, 'view_map', None)
+                reallocated_ins = set([])
+
+                idx_o = 0
+                for out in node.outputs:
+                    ins = None
+                    if dmap and idx_o in dmap:
+                        idx_v = dmap[idx_o]
+                        assert len(idx_v) == 1, "Here we only support the possibility to destroy one input"
+                        ins = node.inputs[idx_v[0]]
+                    if vmap and idx_o in vmap:
+                        assert ins is None
+                        idx_v = vmap[idx_o]
+                        assert len(idx_v) == 1, "Here we only support the possibility to view one input"
+                        ins = node.inputs[idx_v[0]]
+                    if ins:
+                        assert isinstance(ins, theano.Variable)
+                        origin = view_of.get(ins, ins)
+                        view_of[out] = origin
+                        viewed_by[origin].append(out)
+                idx_o += 1                        
+
+                for ins in node.inputs:
+                    assert not (ins in view_of and viewed_by[ins])
+                    if (ins.ndim == 0 and storage_map[ins][0] and ins not in fgraph.outputs and ins.owner):
```

Comment:
I suppose the strange error are fixed by storage_map[ins][0]? At this step, it always contant None. So this condition will always be False. You do not want that.